### PR TITLE
Make consensus module test have separate server node

### LINKED ISSUE #297: Make consensus module test have separate server node

### PR DESCRIPTION
**Pull request by lemonholic**
This resolves #297.

### CODE REVIEW

File: consensus/tests/integration_test.rs [hunk 1]
```diff
@@ -335,3 +332,108 @@ async fn single_server_propose_1() {
...
+
+    let (server_config, other_configs, peers) =
+        setup_server_client_nodes(network_id.clone(), num_nodes - 1).await;
+
+    let block_header = configs_to_block_header(
+        once(&server_config).chain(&other_configs).collect(),
+        voting_powers,
+    );
+
+    // Create active nodes (client nodes) that will be executed by CLI.
```
Comment (junha1) [inline consensus/tests/integration_test.rs:357]: The term 'active' is not meaningful IMO.

File: consensus/tests/integration_test.rs [hunk 2]
```diff
@@ -335,3 +332,108 @@ async fn single_server_propose_1() {
...
+    }
+
+    // Propose the block.
+    let proposer = &mut nodes[0];
+    proposer
+        .set_proposal_candidate(dummy_block_hash, get_timestamp())
+        .await
+        .unwrap();
+
+    // Fetch Progress Sleep Repeat.
```
Comment (junha1) [inline consensus/tests/integration_test.rs:413]: Fetch, progress, sleep, and repeat.

**Review by junha1 — APPROVED** [review]

File: consensus/tests/integration_test.rs [hunk 3]
```diff
@@ -335,3 +332,108 @@ async fn single_server_propose_1() {
...
+            params.clone(),
+            round_zero_timestamp,
+            Some(config.private_key.clone()),
+        )
+        .await
+        .unwrap();
+        nodes.push(consensus);
+    }
+
+    // Create a router node (a server node).
```
[the remark this answers is not in the document]
Comment (lemonholic) [reply consensus/tests/integration_test.rs:373]: ok

[the remark this answers is not in the document]
Comment (lemonholic) [reply consensus/tests/integration_test.rs:373]: fixed: `Create a server node.`

File: consensus/tests/integration_test.rs (hunk 1 above)
Comment (lemonholic) [reply consensus/tests/integration_test.rs:357]: removed it

File: consensus/tests/integration_test.rs (hunk 2 above)
Comment (lemonholic) [reply consensus/tests/integration_test.rs:413]: fixed

**Review by junha1 — APPROVED** [review]

**Pull request merged**